android/src/main/java/com/optimove/reactnative/events/InAppDeeplinkEvent.java

```diff
@@ -25,9 +31,21 @@ public WritableMap getData() {
     WritableMap map = new WritableNativeMap();
 
     map.putInt("messageId", buttonPress.getMessageId());
-    map.putString("deepLinkData", buttonPress.getDeepLinkData().toString());
-    if (buttonPress.getMessageData() != null) {
-      map.putString("messageData", buttonPress.getMessageData().toString());
+
+    try {
+      map.putMap("deepLinkData", JSONtoMapMapper.jsonToReact(buttonPress.getDeepLinkData()));
+    } catch (Throwable e) {
+      Log.e(TAG, String.format("Couldn't parse deepLink data due to: %s", e.getMessage()));
+    }
+
+    if (buttonPress.getMessageData() == null) {
```

Comment:
Should null be written into map here?

According to TS iface it's Record | null https://github.com/optimove-tech/Optimove-SDK-React-Native/blob/dfab86a104c9c1340f7cd18171925a06ea7fce7b/src/inApp.ts#L22

Reply:
(not sure what iOS bridge does)